core/src/main/c/share/ooo.cpp

```diff
@@ -563,12 +555,39 @@ Java_io_questdb_std_Vect_mergeLongIndexesAsc(JAVA_STATIC, jlong pIndexStructArra
             entries[i].size = -1;
         }
     }
-    //this piece allocates memory for merged_index that doesn't show up in stats
-    auto *merged_index = reinterpret_cast<index_t *>(malloc(merged_index_size * sizeof(index_t)));
+    if (merged_index == nullptr) {
```

Comment:
this is unnecessary, method that passes null here is unused

Reply:
fixed